src/super_gradients/training/utils/quantization/core.py

```diff
@@ -0,0 +1,140 @@
+import inspect
+from dataclasses import dataclass
+from enum import Enum
+from typing import Union, Type, Optional
+
+from pytorch_quantization.nn.modules._utils import QuantMixin, QuantInputMixin
+from pytorch_quantization.tensor_quant import QuantDescriptor
+from torch import nn
+
+
+def _extract_init_args(cls, float_instance):
```

Comment:
Are we assuming here that the arguement keys must match the attribute names of the module?

If so, this is can be problematic...In that case we should consider allowing different _extract_init_args that are block specific.

Reply:
We do make that assumption.
This functionality is the default and is used if `SGQuantMixin.from_float` is not overridden.
To do custom stuff, one may override `SGQuantMixin.from_float`